src/gc/impl/conservative/gc.d

```diff
@@ -2012,94 +2022,106 @@ struct Gcx
                     // We don't care abou setting pointsToBase correctly
                     // because it's ignored for small object pools anyhow.
                     auto offsetBase = offset & notbinsize[bin];
-                    biti = offsetBase >> pool.shiftBy;
-                    base = pool.baseAddr + offsetBase;
+                    biti = offsetBase >> Pool.ShiftBy.Small;
                     //debug(PRINTF) printf("\t\tbiti = x%x\n", biti);
 
-                    if (!pool.mark.set(biti) && !pool.noscan.test(biti)) {
-                        stack[stackPos++] = Range(base, base + binsize[bin]);
-                        if (stackPos == stack.length)
-                            break;
+                    if (!pool.mark.set(biti) && !pool.noscan.test(biti))
+                    {
+                        base = pool.baseAddr + offsetBase;
+                        top = base + binsize[bin];
+                        goto LaddRange;
                     }
                 }
                 else if (bin == B_PAGE)
                 {
-                    auto offsetBase = offset & notbinsize[bin];
-                    base = pool.baseAddr + offsetBase;
-                    biti = offsetBase >> pool.shiftBy;
+                    biti = offset >> Pool.ShiftBy.Large;
                     //debug(PRINTF) printf("\t\tbiti = x%x\n", biti);
 
                     pcache = cast(size_t)p & ~cast(size_t)(PAGESIZE-1);
+                    base = cast(void*)pcache;
 
                     // For the NO_INTERIOR attribute.  This tracks whether
                     // the pointer is an interior pointer or points to the
                     // base address of a block.
-                    bool pointsToBase = (base == sentinel_sub(p));
```

Comment:
Was it necessary to remove that name?
Even dmd should be able to optimize this and deciphering `base != sentinel_sub(p)` isn't trivial.

Reply:
> Even dmd should be able to optimize

Unfortunately, it it doesn't just use the comparison in the following `if`, but stores the result to a byte with `sete` and uses that, adding a number of gratuitious instructions.